applegacy/src/main/java/com/hamdan/forzenbook/applegacy/viewmodels/LegacyCreateAccountViewModel.kt

```diff
@@ -0,0 +1,167 @@
+package com.hamdan.forzenbook.applegacy.viewmodels
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hamdan.forzenbook.core.Entry
+import com.hamdan.forzenbook.core.LoginError
+import com.hamdan.forzenbook.createaccount.core.domain.CreateAccountEntrys
+import com.hamdan.forzenbook.createaccount.core.domain.CreateAccountResult
+import com.hamdan.forzenbook.createaccount.core.domain.CreateAccountUseCase
+import com.hamdan.forzenbook.createaccount.core.domain.CreateAccountValidationUseCase
+import com.hamdan.forzenbook.ui.core.R
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LegacyCreateAccountViewModel @Inject constructor(
+    private val createAccountUseCase: CreateAccountUseCase,
+    private val createAccountValidationUseCase: CreateAccountValidationUseCase,
+) : ViewModel() {
+    data class CreateAccountState(
+        val errorId: Int? = null,
+        val firstName: Entry = Entry("", LoginError.NameError.None),
+        val lastName: Entry = Entry("", LoginError.NameError.None),
+        val birthDay: Entry = Entry("", LoginError.BirthDateError.None),
+        val email: Entry = Entry("", LoginError.EmailError.None),
+        val location: Entry = Entry("", LoginError.LocationError.None),
+        val isDateDialogOpen: Boolean = false,
+        val isLoading: Boolean = false,
+    )
+
+    // TODO REMOVE THIS FUNCTION WHEN WORKING ON Legacy Create Account FA-102
+    fun test() {
+        Log.v("Hamdan", "Hello")
+    }
+
+    private var errorId: Int? = null
+    private var firstName: Entry = Entry("", LoginError.NameError.None)
+    private var lastName: Entry = Entry("", LoginError.NameError.None)
+    private var birthDay: Entry = Entry("", LoginError.BirthDateError.None)
+    private var email: Entry = Entry("", LoginError.EmailError.None)
+    private var location: Entry = Entry("", LoginError.LocationError.None)
+    private var isDateDialogOpen: Boolean = false
+    private var isLoading: Boolean = false
+
+    var onCreateAccountUpdate: ((CreateAccountState) -> Unit)? = null
```

Comment:
Why make this a lambda?

Reply:
The intent here is that we're getting away from lambdas, the "easy" update method, and going to MutableStateFlow as is the expected company practice